Cron/Providers/ProcessedOldNotificationsProvider.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2024 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Cron\Providers;
+
+use Adyen\Payment\Api\Repository\AdyenNotificationRepositoryInterface;
+use Adyen\Payment\Helper\Config;
+use Adyen\Payment\Logger\AdyenLogger;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\StoreManagerInterface;
+
+class ProcessedOldNotificationsProvider implements NotificationsProviderInterface
+{
+    public function __construct(
+        private readonly AdyenNotificationRepositoryInterface $adyenNotificationRepository,
+        private readonly SearchCriteriaBuilder $searchCriteriaBuilder,
+        private readonly Config $configHelper,
+        private readonly StoreManagerInterface $storeManager,
+        private readonly AdyenLogger $adyenLogger
+    ) { }
+
+    public function provide(): array
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+        $numberOfDays = $this->configHelper->getRequiredDaysForOldWebhooks($storeId);
+
+        $dateFrom = date('Y-m-d H:i:s', time() - $numberOfDays * 24 * 60 * 60);
+
+        $searchCriteria = $this->searchCriteriaBuilder
+            ->addFilter('done', 1)
+            ->addFilter('processing', 0)
+            ->addFilter('created_at', $dateFrom, 'lteq')
+            ->create();
+
+        try {
+            $items = $this->adyenNotificationRepository->getList($searchCriteria);
+            return $items->getItems();
+        } catch (LocalizedException $e) {
+            $errorMessage = sprintf(
+                __('An error occurred while providing notifications older than %s days!'),
+                $numberOfDays
+            );
+
+            $this->adyenLogger->error($errorMessage);
+
+            return [];
+        }
+    }
+
+    public function getProviderName(): string
```

Comment:
What is this used for?

Reply:
This method mimics [this](https://github.com/Adyen/adyen-magento2/blob/bbc80c7ba86052002e624d0f1c7044631bd831f2/Cron/Providers/OrdersProviderInterface.php#L27) interface even though they are not sharing the same parent. 

The method is not used during the implementation but has been added for future references.

Reply:
Ah for logging I see https://github.com/Adyen/adyen-magento2/blob/main/Cron/CancelOrders.php#L53 maybe we can use it for the same purpose

Reply:
It's a good call! The last commit implements logging using `entityId` and `providerName`. Thanks for the recommendation.